syntax = "proto3";

package proto_catalogue;

message Point {
  double x = 1;
  double y = 2;
}

message RGB {
  uint32 r = 1;
  uint32 g = 2;
  uint32 b = 3;
}

message RGBa {
  uint32 r = 1;
  uint32 g = 2;
  uint32 b = 3;
  double opacity = 4;
}

message Color {
  oneof colors {
    bool no_color = 1;
    string color_name = 2;
    RGB rgb = 3;
    RGBa rgba = 4;
  }
}